crates/uv-configuration/src/vcs_options.rs

```diff
@@ -0,0 +1,123 @@
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+use std::str::FromStr;
+
+use anyhow::{Context, Result};
+use serde::Deserialize;
+
+use uv_fs::Simplified;
+use uv_warnings::warn_user;
+
+/// The version control system to use.
+#[derive(Clone, Copy, Debug, PartialEq, Default, Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+#[cfg_attr(feature = "clap", derive(clap::ValueEnum))]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub enum VersionControl {
+    /// Use Git for version control.
+    #[default]
+    Git,
+
+    /// Do not use version control.
+    None,
+}
+
+impl VersionControl {
+    /// Initializes the VCS system based on the provided path.
+    pub fn init(&self, path: &PathBuf) -> Result<()> {
+        match self {
+            VersionControl::None => Ok(()),
+            VersionControl::Git => Self::init_git(path),
+        }
+    }
+
+    fn init_git(path: &PathBuf) -> Result<()> {
+        let Ok(git) = which::which("git") else {
+            anyhow::bail!("could not find `git` in PATH");
+        };
+
+        if path.join(".git").try_exists()? {
+            warn_user!(
+                "Git repository already exists at `{}`",
+                path.simplified_display()
+            );
+        } else {
+            if !Command::new(git)
+                .arg("init")
+                .current_dir(path)
+                .stdout(Stdio::null())
+                .stderr(Stdio::null())
+                .status()
+                .context("failed to run `git init`")?
+                .success()
+            {
+                anyhow::bail!("`git init` failed at `{}`", path.simplified_display());
+            }
```

Comment:
It seems problematic not to capture and display the output of the command on failure. We have some utilities for this, I think.